### PR TITLE
Implement eachSocket iterator

### LINKED ISSUE #48: Provide method to iterate over currently connected sockets
Ideally there would be something like `server.sockets()` that returns an array of all connected sockets, but generally looking for a way to access the `Socket` instances without having apps keep their own records of who is connected and not.

### PR DESCRIPTION
**Pull request by kpdecker**
Provides a formal API for consumers to interact with currently connected clients.

Fixes #48

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -155,6 +155,15 @@ Sends a message to all the subscribed clients where:
 - `message` - the message sent to the clients. Can be any type which can be safely converted to
   string using `JSON.stringify()`.
 
+
+### `server.eachSocket(callback, options)`
+
+Iterates over all connected sockets, optionally filtering on those that have subscribed to
+a given subscription. This operation is synchronous.
+- `callback` - Iteration callback in the form `function(socket)`.
```
Comment (hueniverse) [inline API.md]: Call this `each` or something other than `callback`.

File: API.md [hunk 2]
```diff
@@ -155,6 +155,15 @@ Sends a message to all the subscribed clients where:
...
+- `options` - Optional options object
+  - `subscription` - Limit to sockets that are subscribed to a given path.
```
Comment (hueniverse) [inline API.md]: Should be more specific: when set to a string path...

File: lib/listener.js [hunk 3]
```diff
@@ -347,6 +347,38 @@ internals.Listener.prototype._subscribe = function (path, socket, next) {
 };
 
 
+internals.Listener.eachSocket = function (callback, options) {
+
```
Comment (hueniverse) [inline lib/listener.js]: Add `options = options || {};` here, then drop the test in the _eachSocket method.

File: API.md (hunk 1 above)
Comment (kpdecker) [inline API.md]: What is the concern here? This matches the `Array.forEach` spec.

**Comment by kpdecker** [comment]
Updated the other two items and pushed.

File: API.md (hunk 1 above)
Comment (hueniverse) [inline API.md]: But not the hapi style guide. callback and next have specific meanings, and are always only called once.

File: lib/listener.js [hunk 4]
```diff
@@ -351,6 +351,40 @@ internals.Listener.prototype._subscribe = function (path, socket, next) {
...
+        return this._sockets.forEach(callback);
+    }
+
+    var sub = this._router.route('sub', subscription);
+    if (sub.isBoom) {
+        return;
+    }
+
+    var route = sub.route;
+    route.subscribers.forEach(sub.paramsArray.length ? subscription : null, function (socket) {     // Filter on path if has parameters
```
Comment (hueniverse) [inline lib/listener.js]: I am not following the first argument here

Comment (kpdecker) [inline lib/listener.js]: This is for filtering to any parameters defined in the subscription path. Similar code exists in the publish flow, but the variable is called `path` there.

**Pull request merged**